Libraries/LibWeb/TrustedTypes/TrustedTypePolicy.cpp

```diff
@@ -182,4 +185,56 @@ WebIDL::ExceptionOr<JS::Value> TrustedTypePolicy::get_trusted_type_policy_value(
     return policy_value;
 }
 
+// https://www.w3.org/TR/trusted-types/#process-value-with-a-default-policy-algorithm
+WebIDL::ExceptionOr<JS::Value> process_value_with_a_default_policy(TrustedTypeName trusted_type_name, JS::Object& global, Variant<GC::Root<TrustedHTML>, GC::Root<TrustedScript>, GC::Root<TrustedScriptURL>, Utf16String> input, InjectionSink sink)
+{
+    auto& vm = global.vm();
+    auto& realm = *vm.current_realm();
+
+    // 1. Let defaultPolicy be the value of global’s trusted type policy factory’s default policy.
+    auto const& default_policy = as<HTML::WindowOrWorkerGlobalScopeMixin>(global).trusted_types()->default_policy();
+
+    // This algorithm routes a value to be assigned to an injection sink through a default policy, should one exist.
+    // FIXME: Open an issue upstream. It is not immediately clear what to do if the default policy does not exist.
+    if (!default_policy)
+        return JS::js_null();
+
+    // 2. Let policyValue be the result of executing Get Trusted Type policy value, with the following arguments:
+    //  defaultPolicy as policy
+    //  stringified input as value
+    //  expectedType’s type name as trustedTypeName
+    //  « trustedTypeName, sink » as arguments
+    auto arguments = GC::RootVector<JS::Value>(vm.heap());
+    arguments.append(JS::PrimitiveString::create(vm, to_string(trusted_type_name)));
+    arguments.append(JS::PrimitiveString::create(vm, to_string(sink)));
+    //  false as throwIfMissing
+    //  3. If the algorithm threw an error, rethrow the error and abort the following steps.
+    auto policy_value = TRY(default_policy->get_trusted_type_policy_value(
+        trusted_type_name,
+        input.visit(
+            [](auto& value) { return value->to_string(); },
+            [](Utf16String& value) { return value; }),
+        arguments,
+        ThrowIfCallbackMissing::No));
+
+    //  4. If policyValue is null or undefined, return policyValue.
+    if (policy_value.is_nullish())
+        return policy_value;
+
+    //  5. Let dataString be the result of stringifying policyValue.
+    auto data_string = TRY(policy_value.to_utf16_string(vm));
+
+    //  6. Return a new instance of an interface with a type name trustedTypeName, with its associated data value set to dataString.
+    switch (trusted_type_name) {
+    case TrustedTypeName::TrustedHTML:
+        return realm.create<TrustedHTML>(realm, move(data_string));
+    case TrustedTypeName::TrustedScript:
+        return realm.create<TrustedScript>(realm, move(data_string));
+    case TrustedTypeName::TrustedScriptURL:
+        return realm.create<TrustedScriptURL>(realm, move(data_string));
+    default:
+        VERIFY_NOT_REACHED();
```

Comment:
If you move the `VERIFY_NOT_REACHED()` outside of the switch, it behaves the same, but the compiler will complain if any new values get added to the enum and they're not handled here.